icu4j/main/classes/core/src/com/ibm/icu/text/DateIntervalFormat.java

```diff
@@ -1963,20 +2047,42 @@ private static String adjustFieldWidth(String inputSkeleton,
         */
 
 
+        int PATTERN_CHAR_BASE = 0x41;
+
         DateIntervalInfo.parseSkeleton(inputSkeleton, inputSkeletonFieldWidth);
         DateIntervalInfo.parseSkeleton(bestMatchSkeleton, bestMatchSkeletonFieldWidth);
+        if (suppressDayPeriodField) {
+            bestMatchIntervalPattern = bestMatchIntervalPattern.replace(" a", "");
+            bestMatchIntervalPattern = bestMatchIntervalPattern.replace("a", "");
+        }
```

Comment:
same comment as in C about this section

Reply:
Fixed.  (The logic here is different from the C++ logic because the C++ code processes the whole pattern string in one pass, while the Java code splits it into lower-bound and upper-bound patterns first and _then_ calls this function.)